config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/AnnotationDeserializer.java

```diff
@@ -56,30 +62,54 @@
-                    if (ele.hasNonNull("id")) {
-                        annotation.setId(ele.get("id").asText());
-                    }
-                    if (ele.hasNonNull("guid")) {
-                        annotation.setGuid(ele.get("guid").asText());
-                    }
-                    if (ele.hasNonNull("description")) {
-                        annotation.setDescription(ele.get("description").asText());
-                    }
-                    if (ele.hasNonNull("tags")) {
-                        final List<String> tags = new ArrayList<>();
-                        ele.get("tags").elements().forEachRemaining(e -> tags.add(e.textValue()));
-                        annotation.setTags(tags);
+                    if (AnnotationConstants.SUPPORTED_ANNOTATION_TYPES.contains(annotation.getType()) && !annotations.contains(annotation)) {
+                        annotation.setName(getNodeAttribute(ele, "name"));
+                        annotation.setId(getNodeAttribute(ele, "id"));
+                        annotation.setGuid(getNodeAttribute(ele, "guid"));
+                        annotation.setDescription(getNodeAttribute(ele, "description"));
+
+                        if (ele.hasNonNull("tags")) {
+                            final List<String> tags = new ArrayList<>();
+                            ele.get("tags").elements().forEachRemaining(e -> tags.add(e.textValue()));
+                            annotation.setTags(tags);
+                        }
+                        annotations.add(annotation);
+                    } else {
+                        alertUser(annotation.getType());
                     }
-                    annotations.add(annotation);
+
                 } else if (ele.isTextual()) {
-                    annotations.add(new Annotation(ele.asText()));
+                    final Annotation annotation = new Annotation(ele.asText());
+                    if (AnnotationConstants.SUPPORTED_ANNOTATION_TYPES.contains(annotation.getType()) && !annotations.contains(annotation)) {
+                        annotations.add(annotation);
+                    } else {
+                        alertUser(annotation.getType());
+                    }
                 }
             });
             return annotations;
         }
         return Collections.emptySet();
     }
+
+    /**
+     * Gets the JsonNode attribute value.
+     * @param node
+     * @param attribute
+     * @return
+     */
+    private String getNodeAttribute(final JsonNode node,  final String attribute) {
+        if (node.hasNonNull(attribute)) {
+            return node.get(attribute).asText();
+        }
+        return EMPTY;
+    }
+
+    /**
+     * Logs the warning for unsupported/duplicate annotation types.
+     * @param annotationType
+     */
+    private void alertUser (final String annotationType) {
+        LOGGER.log(Level.WARNING, "Annotations contain unsupported or duplicate annotation: {0}", annotationType);
```

Comment:
Change it to Found unsupported or duplicate**### d** annotation: {0}

Reply:
Done